internal/service/ec2/status.go

```diff
@@ -312,6 +312,10 @@ func StatusRouteTableAssociationState(conn *ec2.EC2, id string) resource.StateRe
 			return nil, "", err
 		}
 
+		if output.AssociationState == nil {
+			return nil, "", nil
+		}
+
```

Comment:
@blafrisch Thanks for the contribution :tada: :clap:.
This code indicates that an `AssociationState` of `nil` is equivalent to a non-existent or disassociated association - Is that the correct behavior? If there is no `AssociationState` but there is an association, is that indicative of an associated association or not?

Reply:
If there is no `AssociationState` but there is an association it is indicative of an association. Not sure of the best way to represent this as the state is not set.

Reply:
We can do this by returning a "fake" `RouteTableAssociationState` from `StatusRouteTableAssociationState` with a `State` value of "associated".